io/directory.go

```diff
@@ -141,6 +146,22 @@ func (d *BasicDirectory) AddChild(ctx context.Context, name string, node ipld.No
 	return d.node.AddNodeLink(name, node)
 }
 
+// EnumLinksAsync returns a channel which will receive Links in the directory
+// as they are enumerated, where order is not gauranteed
+func (d *BasicDirectory) EnumLinksAsync(ctx context.Context) (<-chan format.LinkResult, error) {
+	linkResults := make(chan format.LinkResult)
+	go func() {
+		defer close(linkResults)
+		for _, l := range d.node.Links() {
+			linkResults <- format.LinkResult{
```

Comment:
This should also select on ctx to avoid leaking goroutines (same in other places)

Reply:
got it. will do.